### PR TITLE
TD-1570 implement remove from self assessment functionality

### PR DESCRIPTION
**Pull request by sherif-olaboye**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-1570

### Description
Adding RemoveDelegateSelfAssesment page to user to confirm the action and adding a query which update the status of the delegate from one to two

### Screenshots
image
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <partial name="Shared/_NavMenuItems" />
+}
+
+<div class="nhsuk-grid-row">
+    <div class="nhsuk-grid-column-full word-break">
+        @if (errorHasOccurred)
+        {
+            <vc:error-summary order-of-property-names="@(new[] { nameof(Model.ActionConfirmed) })" />
+        }
+        <h1 id="page-heading">Confirm Remove Delegate Self Assessment</h1>
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml:20]: Change this text to "Remove from self assessment"

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        <div class="nhsuk-grid-row">
+            <div class="nhsuk-grid-column-one-quarter nhsuk-heading-l">
+                <div class="nhsuk-u-font-weight-bold">
+                    Delegate self assessment:
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml:24]: this heading should be just "Delegate:" not "Delegate self assessment:"

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                </div>
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml:25]: Have a separate row for "Self Assessment:" and load the self assessment name over there. See the screen for remove from course for this.

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        </div>
+
+        <form method="post" asp-controller="ActivityDelegates">
+            <div class="nhsuk-checkboxes__item">
+                <vc:single-checkbox asp-for="@nameof(Model.ActionConfirmed)"
+                                    label="I am sure that I wish to remove @Model.FirstName @Model.LastName from self assessment"
+                                    hint-text="This action will remove the delegate from your self assessment list and you will no longer be able to view this delegate from self assessments." />
+            </div>
+            <button type="submit" class="nhsuk-button nhsuk-button--danger nhsuk-u-margin-top-4" asp-action="RemoveDelegateSelfAssessment">
+                Remove
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml:39]: Change button name to "Remove from self assessment"

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/SelfAssessmentDataService.cs [hunk 5]
```diff
@@ -333,5 +337,32 @@ SELECT	s.DelegateUserID
             );
             return (delegateUserCard, ResultCount);
         }
+        public RemoveSelfAssessmentDelegate GetDelegateSelfAssessmentByCandidateAssessmentsId(int candidateAssessmentsId)
+        {
+            return connection.QueryFirstOrDefault<RemoveSelfAssessmentDelegate>(
+                 @"Select
+                    ca.Id AS CandidateAssessmentsId,
+                    ca.SelfAssessmentID,
+                    da.FirstName_deprecated AS FirstName,
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/SelfAssessmentDataService.cs:346]: the first name and last name columns are deprecated and should not be used. Please fetch the name from Users table

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml (hunk 1 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml]: Done

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml (hunk 2 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml]: Done

File: DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml (hunk 4 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/ActivityDelegates/RemoveDelegateSelfAssessment.cshtml]: Done

File: DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/SelfAssessmentDataService.cs (hunk 5 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/SelfAssessmentDataService.cs]: Done

**Pull request merged**